src/testing/basic_test.rs

```diff
@@ -1,390 +1,230 @@
-                    &quorum_proposal.block_header,
-                );
-            let payload_builder_commitment =
-                <TestBlockHeader as BlockHeader<TestTypes>>::builder_commitment(
-                    &quorum_proposal.block_header,
-                );
+        prev_quorum_proposal = Some(quorum_proposal.clone());
 
-            let qc_signature = <TestTypes as hotshot_types::traits::node_implementation::NodeType>::SignatureKey::sign(
+        let qc_signature = <TestTypes as hotshot_types::traits::node_implementation::NodeType>::SignatureKey::sign(
                         &private_key,
-                        payload_vid_commitment.as_ref(),
+                        block_vid_commitment.as_ref(),
                         ).expect("Failed to sign payload commitment while preparing QC proposal");
 
-            let sqc_msg = QuorumProposalMessage::<TestTypes> {
+        let quorum_proposal_msg =
+            MessageType::QuorumProposalMessage(QuorumProposalMessage::<TestTypes> {
                 proposal: Arc::new(Proposal {
                     data: quorum_proposal.clone(),
                     signature: qc_signature,
                     _pd: PhantomData,
                 }),
                 sender: pub_key,
-            };
-
-            // Prepare the decide message
-            let leaf = match i {
-                0 => {
-                    Leaf::genesis(
-                        &TestValidatedState::default(),
-                        &TestInstanceState::default(),
-                    )
-                    .await
-                }
-                _ => {
-                    let block_payload = BlockPayload::<TestTypes>::from_bytes(
-                        &encoded_transactions,
-                        <TestBlockHeader as BlockHeader<TestTypes>>::metadata(
-                            &quorum_proposal.block_header,
-                        ),
-                    );
-                    let mut current_leaf = Leaf::from_quorum_proposal(&quorum_proposal);
-                    current_leaf
-                        .fill_block_payload(block_payload, TEST_NUM_NODES_IN_VID_COMPUTATION)
-                        .unwrap();
-                    current_leaf
-                }
-            };
-
-            let sdecide_msg = DecideMessage::<TestTypes> {
-                latest_decide_view_number: leaf.view_number(),
-            };
+            });
 
-            // validate the signature before pushing the message to the builder_state channels
-            // currently this step happens in the service.rs, wheneve we receiver an hotshot event
-            tracing::debug!("Sending transaction message: {:?}", tx);
-            for res in handle_received_txns(
-                &senders.transactions,
-                vec![tx.clone()],
-                TransactionSource::HotShot,
-            )
+        // send quorum and DA proposals for this round
```

Comment:
I thought the constructed proposals were associated with the previous view, but according to this comment, they are for the current view. If that's the case, shouldn't we use the new transactions rather than `prev_proposed_transactions` to construct proposals?

Reply:
Oh is it because the proposals for _this_ round are with the _previous_ view number? "Round" and "view" are different I guess.

Reply:
Yeah, this round's proposals are emulating a block built from txns received in previous round being proposed.